docs/synonyms.md

```diff
@@ -1,5 +1,180 @@
+
+    ```json
+    {
+        "analysis": {
+            "synonym_sources": {
+                "english": {
+                    "collection": "en_thesaurus",
+                    "analyzer": "en"
+                },
+                "german": {
+                    "collection": "de_thesaurus",
+                    "analyzer": "de"
+                }
+            }
+        }
+   }
+   ```
+
+    There are two synonym sources named "english" and "german," each associated with its respective "synonym collection" and analyzer. In any text field mapping, a synonym source can be specified to enable synonym expansion when the field is queried. The analyzer of the synonym source must match the analyzer of the field mapping to which it is applied.
+
+* Any text-based Bleve query (e.g., match, phrase, term, fuzzy, etc.) will use the synonym source (if available) for the queried field to expand the search terms using the thesaurus created from user-defined synonym definitions. The behavior for specific query types is as follows:
+
+    1. Queries with `fuzziness` parameter: For queries like match, phrase, and match-phrase that support the `fuzziness` parameter, the queried terms are fuzzily matched with the thesaurus's LHS terms to generate candidate terms. These terms are then combined with the results of fuzzy matching against the field dictionary, which contains the terms present in the queried field.
+
+    2. Wildcard, Regexp, and Prefix queries: These queries follow a similar approach. First, the thesaurus is used to expand terms (e.g., LHS terms that match the prefix or regex). The resulting terms are then combined with candidate terms from dictionary expansion.
+
+## Indexing
+
+Below is an example of using the Bleve API to define synonym sources, index synonym definitions, and associate them with a text field mapping:
+
+```go
+// Define a document to be indexed.
+doc := struct {
+    Text string `json:"text"`
+}{
+    Text: "hardworking employee",
+}
+
+// Define a synonym definition where "hardworking" has equivalent terms.
+synDef := &bleve.SynonymDefinition{
+    Synonyms: []string{
+        "hardworking",
+        "industrious",
+        "conscientious",
+        "persistent",
+    },
+}
+
+// Define the name of the `synonym collection`.
+// This collection groups multiple synonym definitions.
+synonymCollection := "collection1"
+
+// Define the name of the `synonym source`.
+// This source will be associated with specific field mappings.
+synonymSourceName := "english"
+
+// Define the analyzer to process terms in the synonym definitions.
+// This analyzer must match the one applied to the field using the synonym source.
+analyzer := en.AnalyzerName
+
+// Configure the synonym source by associating it with the synonym collection and analyzer.
+synonymSourceConfig := map[string]interface{}{
+    "collection": synonymCollection,
+    "analyzer":   analyzer,
+}
+
+// Create a new index mapping.
+bleveMapping := mapping.NewIndexMapping()
```

Comment:
bleve.NewIndexMapping* right? (The API is defined in the bleve package itself, which is what the users are recommended to use I think?)